logger/logger.go

```diff
@@ -33,13 +36,29 @@ func (t *telegrafLog) Write(b []byte) (n int, err error) {
 	return t.writer.Write(line)
 }
 
+func (t *telegrafLog) Close() error {
+	closer, isCloser := t.writer.(io.Closer)
```

Comment:
Indeed, I'll keep the underlying writer as a separate field in the struct, so it can be closed. 
I need to close the io.file to be able to delete lingering temp files in this test: https://github.com/influxdata/telegraf/blob/cd921626b42eec0beea2ec099197f6ec6515ee15/logger/logger_test.go#L94

Reply:
I am most concerned about potentially logging an error when everything is fine.  If you do close the writer make sure that it still works after a SIGHUP, on Linux this signal will reload the config file.  I think it will be fine though because no one will call this outside of the tests.

Reply:
Yes, this only used in the tests to be able to remove lingering test files, so it won't be a problem at runtime.